net/api/test_server.cc.h

```diff
@@ -9,17 +9,216 @@
 #ifndef BRICKS_NET_API_TEST_SERVER_CC_H
 #define BRICKS_NET_API_TEST_SERVER_CC_H
 
-#include "../../dflags/dflags.h"
+#include <thread>
 
+#include "api.h"
+#include "../url/url.h"
 #include "../http/test_server.cc.h"
 
+#include "../../strings/printf.h"
+
 namespace bricks {
 namespace net {
 namespace api {
 namespace test {
 
-struct TestHTTPServer_APIImpl : bricks::net::http::test::TestHTTPServer_HTTPImpl {
-  // TODO(dkorolev): Real API implementation goes here.
+enum class Method { UNINITIALIZED, GET, POST };
+
+struct Request {
+  const url::URL& url;
+  HTTPServerConnection& connection;
+  const HTTPReceivedMessage& message;
+};
+
+struct Routes {
```

Comment:
We'll need to come up with a name :-)

I like neither handler (too broad) nor locations (too web-oriented). Route-based handlers is what sounds best to me so far, but I'll be flexible.